src/core/components/start.js

```diff
@@ -56,9 +60,15 @@ module.exports = (self) => {
         self._bitswap = new Bitswap(
           self.libp2p,
           self._repo.blocks,
+          self._provider,
           { statsEnabled: true }
         )
 
+        if (!get(self._options, 'offline') &&
+          (get(self._options, 'libp2p.config.dht.enabled', false) || get(self._options, 'libp2p.modules.contentRouting', false))) {
+          self._provider.start()
```

Comment:
Because `Provider.start()` is async, bitswap may start before provider finishes starting. It may be safer to wait till `Provider.start()` has completed before starting bitswap

Reply:
I switched the provider start to the previous function of the `series` array, and "callbackifyed" it to guarantee we wait for it

Reply:
The logic makes sense. I would suggest moving all the new code to another entry in the `series` array

Reply:
I can't move the Provider instantiation, as I would have to get the config again, or keeping it in a temporary constant. I would prefer to keep it there, and move the starting code to the new entry of the series array

Reply:
Ok makes sense